testcases/open_posix_testsuite/conformance/interfaces/strcpy/1-1.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * This file is licensed under the GPL license. For the full content
+ * of this license, see the COPYING file at the top level of this
+ * source tree.
+ */
+
+/*
+ * assertion:
+ *	The function will copy the string pointed to by s1 (including the
+ *	terminating NUL character) into the array pointed to by s2. And
+ *	it will return s2.
+ *
+ * method:
+ *	-Define a sample string s1.
+ *	-Use strcpy() to copy s1 into sample string s2.
+ *	-Compare both strings(s1 and s2).
+ *	-Also compare returned pointer with s2.
+*/
+
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#include "../testfrmw/testfrmw.h"
+#include "../testfrmw/testfrmw.c"
+
+#define TNAME "strcpy/1-1.c"
+
+int main(void)
+{
+	static char sample_str_1[] = "HELLO WORLD";
+	char sample_str_2[sizeof(sample_str_1)] = { };
+	char *ret_str = NULL;
+
+	ret_str = strcpy(sample_str_2, sample_str_1);
+
```

Comment:
Here as well, we really lack a coverage if we copy a single pre-defined string, we should loop over different string lengths starting from zero length string up to some reasonable size at least.

Reply:
Looking at the existing LTP test suite, i thought checking for one scenario is enough for conformance testing. And also "reasonable size" is more like subjective, Can you please be more specific, that will be really helpful.

Reply:
A lot of code in LTP is mess and we are cleaning up and fixing it as fast as we can, but given the size of the project it will take years to finish. So for the time being most of the code cannot be taken as good example, some of the tests does not make much sense or are not really doing anything useful...

Hence for new tests, you should really think how to exercise the call best before you even start to write the code. Think of a special cases etc. The strcpy() call is pretty simple though, so what we should do is to exercise it with wide range of valid inputs. The optimized variants of strcpy() are written in assembly and may call different pieces of code for different lengths, so it may make sense to loop over sizes starting from empty string up to a few kilobytes.